aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/mail/MailStepTest.java

```diff
@@ -0,0 +1,96 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins.workflow.steps.mail;
+
+import hudson.model.Result;
+import hudson.model.queue.QueueTaskFuture;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.flow.FlowExecution;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.mock_javamail.Mailbox;
+
+import javax.mail.Message;
+import javax.mail.internet.MimeMultipart;
+import java.io.IOException;
+
+/**
+ * @author <a href="mailto:tom.fennelly@gmail.com">tom.fennelly@gmail.com</a>
+ */
+public class MailStepTest {
+
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    @Test
+    public void test_missing_param_no_continue() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "workflow");
+        // leave out he subject and body
+        job.setDefinition(new CpsFlowDefinition("node {" +
+                "   mail(to: 'tom.abcd@jenkins.org');" +
+                "}", true));
+
+        QueueTaskFuture<WorkflowRun> workflowRunQueueTaskFuture = job.scheduleBuild2(0);
+        workflowRunQueueTaskFuture.waitForStart();
+        WorkflowRun run = jenkinsRule.assertBuildStatus(Result.FAILURE, workflowRunQueueTaskFuture.get());
+
+        FlowExecution execution = run.getExecution();
+        Assert.assertEquals("Email not sent. All mandatory properties must be supplied ('to', 'subject', 'body').", execution.getCauseOfFailure().getMessage());
+    }
+
+    @Test
+    public void test_missing_param_has_continue() throws Exception {
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "workflow");
+        // leave out he subject and body
+        job.setDefinition(new CpsFlowDefinition("node {" +
+                "   mail(to: 'tom.abcd@jenkins.org', continueOnError: true);" +
+                "}", true));
+
+        WorkflowRun run = jenkinsRule.assertBuildStatusSuccess(job.scheduleBuild2(0));
+        Assert.assertTrue(run.getLog(1000).toString().contains("Running: Mail, ERROR: Email not sent. All mandatory properties must be supplied ('to', 'subject', 'body')."));
+    }
+
+    @Test
+    public void test_send() throws Exception {
+        Mailbox.clearAll();
+
+        WorkflowJob job = jenkinsRule.jenkins.createProject(WorkflowJob.class, "workflow");
+        job.setDefinition(new CpsFlowDefinition("node {" +
```

Comment:
Should be no need to wrap it in `node`; it does not require a `FilePath` or other node-related context parameters.